exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdSelectivity.java

```diff
@@ -356,8 +426,8 @@ private boolean isMultiColumnPredicate(final RexNode node) {
     return findAllRexInputRefs(node).size() > 1;
   }
 
-  private static List<RexInputRef> findAllRexInputRefs(final RexNode node) {
-      List<RexInputRef> rexRefs = new ArrayList<>();
+  private static Set<RexInputRef> findAllRexInputRefs(final RexNode node) {
```

Comment:
Would this not break the existing logic? For a predicate like $0=$0 using the `Set` would cause the `isMultiColumnPredicate` function to return false.

Reply:
Yes, thanks for pointing that out, even though that predicate $0=$0 is unexpected (something to investigate in future).  I have reverted this change and it returns a List as before.  Instead, now where the original call to `isMultiColumnPredicate()` happens I added a second condition (line 182) that ensures that conditions of type `$1 > 10 AND $1 < 20` which are created after calling `preProcessRangeConditions()` are not considered the same as multicolumn predicates.